lib/Twig/Extension/Core.php

```diff
@@ -901,9 +923,25 @@ function twig_in_filter($value, $compare)
     } elseif (is_string($compare) && (is_string($value) || is_int($value) || is_float($value))) {
         return '' === $value || false !== strpos($compare, (string) $value);
     } elseif ($compare instanceof Traversable) {
-        return in_array($value, iterator_to_array($compare, false), is_object($value) || is_resource($value));
+        if (is_object($value) || is_resource($value)) {
```

Comment:
Advantage here is not copying the data from the Traversable object to an array and then start searching for the item.